erpnext/selling/doctype/sales_order/sales_order.py

```diff
@@ -541,9 +541,7 @@ def validate_supplier_after_submit(self):
 
 		for item in self.items:
 			if item.supplier:
-				supplier = frappe.db.get_value(
-					"Sales Order Item", {"parent": self.name, "item_code": item.item_code}, "supplier"
-				)
+				supplier = frappe.db.get_value("Sales Order Item", item.name, "supplier")
```

Comment:
How about checking `item.has_value_changed("supplier")` instead?

Reply:
Because `has_value_changed` does not work well for child tables:

![image](https://github.com/frappe/erpnext/assets/10946971/7a882e69-e4ba-4c01-b1a9-8f2e12f984e5)

---

I would have to call load_doc_before_save **on children too**, which is not better in terms of performance.

![image](https://github.com/frappe/erpnext/assets/10946971/dfc4c1a2-0ff7-4889-a33a-471a01b02718)